clippy_lints/src/unused_io_amount.rs

```diff
@@ -130,23 +170,46 @@ fn should_lint<'a>(cx: &LateContext<'a>, mut inner: &'a hir::Expr<'a>) -> Option
     check_io_mode(cx, inner)
 }
 
-fn pattern_is_ignored_ok(cx: &LateContext<'_>, pat: &hir::Pat<'_>) -> bool {
+enum OkPatternKind {
```

Comment:
This could be removed and all instances removed by booleans, and all checking would be easier.